sale_saleor/models/saleor_account.py

```diff
@@ -0,0 +1,4300 @@
+    )
+
+    verify_ssl = fields.Boolean(default=True)
+
+    token = fields.Char(readonly=True)
+    token_expiry = fields.Datetime(readonly=True)
+    active = fields.Boolean(default=False)
+
+    # Batch size for queue jobs when syncing multiple records
+    job_batch_size = fields.Integer(default=10, string="Batch Size")
+
+    # App/Webhook automation fields
+    saleor_app_id = fields.Char(copy=False, readonly=True, string="Saleor App ID")
+    saleor_app_token = fields.Char(copy=False, readonly=True)
+    saleor_customer_webhook_id = fields.Char(
+        copy=False, readonly=True, string="Saleor Webhook ID"
+    )
+    saleor_payment_webhook_id = fields.Char(
+        copy=False, readonly=True, string="Saleor Payment Webhook ID"
+    )
+    saleor_webhook_secret = fields.Char(
+        copy=False,
+        help="HMAC secret used to verify incoming webhook payloads from Saleor",
+    )
+    saleor_order_webhook_id = fields.Char(
+        copy=False, readonly=True, string="Saleor Order Webhook ID"
+    )
+
+    def _get_client(self):
+        self.ensure_one()
+        # Prefer App token if available (long-lived). Otherwise use staff JWT.
+        client = SaleorClient(self.base_url, verify_ssl=self.verify_ssl)
+        if self.saleor_app_token:
+            client.set_token(self.saleor_app_token)
+        else:
+            self._refresh_token(client)
+        return client
+
+    @api.depends("odoo_base_url")
+    def _compute_webhook_url(self):
+        """Compute webhook target URLs based on the Odoo base URL."""
+        for rec in self:
+            if rec.odoo_base_url:
+                base = rec.odoo_base_url.rstrip("/")
+                rec.customer_webhook_url = base + "/saleor/webhook/customer"
+                rec.payment_webhook_url = base + "/saleor/webhook/order_payment"
+                rec.order_webhook_url = base + "/saleor/webhook/order_created_updated"
+            else:
+                rec.customer_webhook_url = False
+                rec.payment_webhook_url = False
+                rec.order_webhook_url = False
+
+    @api.constrains("active")
+    def _check_single_active_account(self):
+        for rec in self:
+            if rec.active:
+                count_active = self.search_count([("active", "=", True)])
+                if count_active > 1:
+                    raise UserError(
+                        self.env._("Only one Saleor account can be active at a time.")
+                    )
+
+    def _refresh_token(self, client=None):
+        self.ensure_one()
+        client = client or SaleorClient(self.base_url, verify_ssl=self.verify_ssl)
+        token = client.token_create(self.email, self.password)
+        self.write(
+            {
+                "token": token,
+                # Saleor default JWT often 5min by default; set conservative 4m
+                "token_expiry": fields.Datetime.now() + timedelta(minutes=4),
+            }
+        )
+        client.set_token(token)
```

Comment:
@Kimkhoi3010: this is calling `token_create` too much which make it hit their auth api so much,
Since we stored `token_expiry` here, so we can check if current time > token_expiry: 4m, then we should call `token_create`. Otherwise, we could use stored token for this work